src/include/mapping.h

```diff
@@ -50,6 +50,7 @@
 #define MAPPING_MAYSHARE	0x100000
 #define MAPPING_NOOVERLAP	0x200000
 #define MAPPING_COPYBACK	0x400000
+#define MAPPING_LOG_DIRTY	0x800000
```

Comment:
Or alternatively this flag can be made generic,
so that the mapping code itself has an API to
track dirty pages, like page_is_dirty() etc, and
vgaemu can use such API instead of forcing the
protect/unprotect calls. Mapping code can then
decide whether to use the protection or use the
KVM API. But this is probably quite a bit of work.